.github/workflows/benchmark.yml

```
@@ -0,0 +1,33 @@
+name: Benchmark
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+permissions:
+  contents: write
+  deployments: write
+env:
+  DEFAULT_GO_VERSION: 1.17
```

```suggestion
  DEFAULT_GO_VERSION: 1.16
```

1.16 is our current MSV and we run our primary test suites against it.  We should probably use it here too.